tron/core/jobrun.py

```diff
@@ -278,17 +280,33 @@ def cleanup(self):
     def get_action_run(self, action_name):
         return self.action_runs.get(action_name)
 
-    def log_state_update(self, state, action_name=None):
+    def log_state_update(self, state: str, action_name: Optional[str] = None) -> None:
         if action_name is None:
             state = f"job_{state}"
         else:
             action_name = str(action_name)
 
+        # Tron currently manges two major types of workloads: batch jobs and Spark jobs
+        # and the team that manages Spark would like to be able to monitor the runtime
+        # of Spark actions in Tron at an aggregate level. We currently ingest these state updates
+        # into various systems and have dashboards on this data, but it's currently non-trivial to
+        # separate out what actions belong to which workload without tagging these state updates
+        executor = None  # we log job state updates with this function - which don't have an executor
+        if self.action_graph and action_name and self.action_graph.action_map.get(action_name):
+            action = self.action_graph.action_map[action_name]
+            # TODO: replace this with checking if the action executor is spark once we're running
+            # the spark driver in k8s
+            if "spark-submit" in action.command:
+                executor = "spark"
```

Comment:
won't every paasta spark-run have spark-submit in the command as well?

Reply:
yeah I think it's the case where every paasta spark-run would have spark-submit inside, but not every command that has spark-submit would have paasta spark-run (although this shouldn't really be the cause in soa-configs) because that would probably mean that they wouldn't really be able to use the spark pools

Reply:
cool cool, it sounded like @88manpreet wanted to track all spark, so this is probably fine then :)